pkg/repotool/repotool.go

```diff
@@ -1,31 +1,138 @@
 package repotool
 
 import (
+	"errors"
+	"fmt"
 	"os"
+	"os/exec"
+	"regexp"
+	"strings"
 
-	"github.com/go-git/go-git/v5"
+	"github.com/grafana/plugin-validator/pkg/logme"
 )
 
-func CloneToTempDir(uri string) (string, func(), error) {
+type GitUrl struct {
+	BaseUrl string
+	Ref     string
+	RootDir string
+}
+
+func CloneToTempDir(uri string, ref string) (string, func(), error) {
+
+	var err error
+
+	err = checkDependencies()
+	if err != nil {
+		return "", nil, err
+	}
+
 	// create a tmp dir
 	tmpDir, err := os.MkdirTemp("", "validator")
 	if err != nil {
 		return "", nil, err
 	}
-	_, err = git.PlainClone(tmpDir, false, &git.CloneOptions{
-		URL:      uri,
-		Progress: os.Stdout,
-		Depth:    1, // only clone the latest commit
-	})
+
+	// construct command to clone
+	cmd := []string{"git", "clone", "--depth", "1", uri, tmpDir}
```

Comment:
Moved from go-git to calling system command since go-git doesn't support an easy clone of a ref without specifying if it is a commit, a tag or a branch.